libretro/libretro.cpp

```diff
@@ -1076,12 +1113,38 @@ bool8 S9xDeinitUpdate(int width, int height)
       }
    }
 
+
+	 if(width==MAX_SNES_WIDTH && hires_blend)
+	 {
+#define AVERAGE_565(el0, el1) (((el0) & (el1)) + ((((el0) ^ (el1)) & 0xF7DE) >> 1))
+      for (register int y = 0; y < height; y++)
+			{
+        register uint16 *input = (uint16 *) ((uint8 *) GFX.Screen + y * GFX.Pitch);
+        register uint16 *output = (uint16 *) ((uint8 *) GFX.Screen + y * GFX.Pitch);
+				register uint16 l, r;
+
+				l = 0;
+				for (register int x = 0; x < (width >> 1); x++)
+				{
+						r = *input++;
+						*output++ = AVERAGE_565 (l, r);
+						l = r;
+
+						r = *input++;
+						*output++ = AVERAGE_565 (l, r);
+						l = r;
+				}
+			}
+	 }
+
+
    video_cb(GFX.Screen, width, height, GFX.Pitch);
    return TRUE;
 }
 
 bool8 S9xContinueUpdate(int width, int height)
 {
+   S9xDeinitUpdate(width, height);
```

Comment:
Doing this in ContinueUpdate will update with only half the scanlines ready. Previously that produced corrupted images for example in ASP. That no longer happens in the latest version, so I guess we can take this in, but it still might produce unwanted effects in some games.